src/compiler/transformers/es6.ts

```diff
@@ -2614,8 +2611,8 @@ namespace ts {
             // Only substitute the identifier if we have enabled substitutions for block-scoped
             // bindings.
             if (enabledSubstitutions & ES6SubstitutionFlags.BlockScopedBindings) {
-                const original = getOriginalNode(node);
-                if (isIdentifier(original) && !nodeIsSynthesized(original) && original.parent && isNameOfDeclarationWithCollidingName(original)) {
+                const original = getSourceTreeNodeOfType(node, isIdentifier);

```

Comment:
There's no reason why not. `getSourceTreeNodeOfType` and `getSourceTreeNode` follow `original` pointers, but return `undefined` if the original node is not a source tree node (e.g. the node must have a `parent` or be a `SourceFile`.)